controllers/buildpipeline/buildpipeline_controller.go

```diff
@@ -104,7 +104,13 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	if err != nil {
 		logger.Error(err, "Failed to get Application from Component",
 			"Component.Name ", component.Name, "Component.Namespace ", component.Namespace)
-		return ctrl.Result{}, err
+		tknErr := tekton.AnnotateBuildPipelineRunWithCreateSnapshotAnnotation(ctx, pipelineRun, r.Client, err)
+		if tknErr != nil {
+			return ctrl.Result{}, tknErr
+		}
+		if errors.IsNotFound(err) {
+			return ctrl.Result{}, err
+		}
```

Comment:
This is wrong it doesn't do what ticket is asking for, it continues with reconciliation
```suggestion
		if errors.IsNotFound(err) {
			return helpers.HandleLoaderError(logger, err, "application", "component")
		}
		return ctrl.Result{}, err
```